component/admin/models/stockrooms.php

```diff
@@ -0,0 +1,133 @@
+ * @license     GNU General Public License version 2 or later; see LICENSE
+ */
+
+defined('_JEXEC') or die;
+
+/**
+ * Model Stockrooms
+ *
+ * @package     RedSHOP.Backend
+ * @subpackage  Model
+ * @since       __DEPLOY_VERSION__
+ */
+class RedshopModelStockrooms extends RedshopModelList
+{
+	protected $_table_prefix = '#__redshop_';
+
+	/**
+	 * Construct class
+	 *
+	 * @since __DEPLOY_VERSION__
+	 */
+	public function __construct($config = array())
+	{
+		if (empty($config['filter_fields'])) {
+			$config['filter_fields'] = array(
+				'id',
+				'name'
+			);
+		}
+
+		parent::__construct($config);
+	}
+
+	/**
+	 * Method to auto-populate the model state.
+	 *
+	 * This method should only be called once per instantiation and is designed
+	 * to be called on the first call to the getState() method unless the model
+	 * configuration flag to ignore the request is set.
+	 *
+	 * Note. Calling getState in this method will result in recursion.
+	 *
+	 * @param string $ordering  An optional ordering field.
+	 * @param string $direction An optional direction (asc|desc).
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function populateState($ordering = 'id', $direction = 'asc')
+	{
+		$search = $this->getUserStateFromRequest($this->context . '.filter.search', 'filter_search');
+		$this->setState('filter.search', $search);
+
+		// List state information.
+		parent::populateState($ordering, $direction);
+	}
+
+	/**
+	 * Method to get a store id based on model configuration state.
+	 *
+	 * This is necessary because the model is used by the component and
+	 * different modules that might need different sets of data or different
+	 * ordering requirements.
+	 *
+	 * @param string $id A prefix for the store id.
+	 *
+	 * @return  string  A store id.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function getStoreId($id = '')
+	{
+		$id .= ':' . $this->getState('filter.search');
+
+		return parent::getStoreId($id);
+	}
+
+	/**
+	 * Method to build an SQL query to load the list data.
+	 *
+	 * @return      string  An SQL query
+	 */
+	protected function getListQuery()
+	{
+		$db    = JFactory::getDbo();
+		$query = $db->getQuery(true)
+			->select('*')
+			->from($db->qn('#__redshop_stockroom'));
+
+		$search = $this->getState('filter.search');
+
+		if ($search) {
+			$query->where(
+				$db->qn('name') . ' LIKE ' . $db->q('%' . $search . '%') . ' OR '
+			);
+		}
+
+		// Add the list ordering clause.
+		$orderCol  = $this->state->get('list.ordering', 'id');
+		$orderDirn = $this->state->get('list.direction', 'asc');
+
+		$query->order($db->escape($orderCol . ' ' . $orderDirn));
+
+		return $query;
+	}
+
+	public function frontpublish($cid = array(), $publish = 1)
+	{
+		if (count($cid)) {
+			$cids = implode(',', $cid);
+
+			$query = 'UPDATE ' . $this->_table_prefix . 'stockroom'
+				. ' SET `show_in_front` = ' . intval($publish)
+				. ' WHERE id IN ( ' . $cids . ' )';
```

Comment:
@hoang-tran-redweb please change to query object

Reply:
I fixed. @hungnguyenckc